api/watch/plan.go

```diff
@@ -39,22 +46,20 @@ func (p *Plan) RunWithConfig(address string, conf *consulapi.Config) error {
 		return fmt.Errorf("Failed to connect to agent: %v", err)
 	}
 
-	// Create the logger
-	output := p.LogOutput
-	if output == nil {
-		output = os.Stderr
-	}
-	logger := log.New(output, "", log.LstdFlags)
-
 	return p.RunWithClientAndLogger(client, logger)
 }
 
 // RunWithClientAndLogger runs a watch plan using an external client and
 // log.Logger instance. Using this, the plan's Datacenter, Token and LogOutput
 // fields are ignored and the passed client is expected to be configured as
 // needed.
-func (p *Plan) RunWithClientAndLogger(client *consulapi.Client,
-	logger *log.Logger) error {
+func (p *Plan) RunWithClientAndLogger(client *consulapi.Client, logger hclog.Logger) error {
```

Comment:
You can't change the signature of this method as it would break backwards compatibility in the public `api/watch` package. You'll have to make a new method.